src/utils/bitfield.rs

```diff
@@ -0,0 +1,231 @@
+use core::ops::{Bound, Range, RangeBounds};
+
+/// Trait for manipulating subsets of integers.
+pub trait BitField {
+    const TYPE_SIZE: usize;
```

Comment:
Can't you infer this size from self's size using `core::mem::size_of` ?

Reply:
Done in e42f3bc9a7b4400e0048f98fe52a3cfd332ce0a4